### ISSUE TITLE
Special hidden parameters

### ISSUE THREAD

**Issue by rebajz** [Codeception/Codeception#3408]
Many time I need doing some test , where I must add my login and password (for example for Gmail).  Right now I must doing this test on my local machine (i cannot share my login with other testers). 

My idea... 
Please enable this new option in codeception. 

I will like write my login and password to command line as parametr, something as:  codecept run --env myStagePhantom acceptance mytestCest.php --html --login myLogin --password myPassword

In test will used this login and password in hidden mode (on screen and html report and other file)  will  visible only text ****\* not myPassword.

**Comment by nordbergr** [comment]
would love to know if anything came of this.

**Comment by Naktibalda** [comment]
Nothing yet.

**Comment by DavertMik** [comment]
I don't plan to implement this in nearest future. If @nordbergr or @rebajz or anyone else is interested I can guide you so you could make a PR. However, I'm still not sure of the best way to implement it.

**Comment by kryzalid87** [comment]
@DavertMik If you could create a guide for hiding passwords or other fields I/We will be thankful :) That will be a good future in some cases

**Comment by edno** [comment]
Instead of passing parameters in command line, I would suggest to extend the [Parameters](http://codeception.com/docs/06-ModulesAndHelpers#Dynamic-Configuration-With-Parameters) feature and exposing the parameters as a [current value](http://codeception.com/docs/07-AdvancedUsage#Current-values).
So the usage will be like:
```php
$params = $scenario->current('params');
$login = $params['login']
```
This way you can store this information locally in a file that is not part of your repository.

Another option using `Parameters` would be the support of inline parameters, and therefore it would be possible to use the `-o` overriding option in command line for passing the parameters, like:
```shell
codecept run -o "params:    login: mysecretlogin    password:s3cr3Tp4ssw0rD"
```

**Comment by kryzalid87** [comment]
@edno - but that`s not the problem. We can as well use system envs in code but still if application will run with -vvv flag for better readability, or there will be an error, we will see inserted password. What I/We want to do is to be able to hide specific text. Another example is when I use JQuery to access elements, and ther`s a lot of unecesarry outputs - we add comments for people to know whats going on. I was looking on a codeception code but after few hours I didn`t find the right place.

**Comment by edno** [comment]
@kryzalid87, so more or less you want to intercept Codeception output and obfuscate/hide some information?

**Comment by nordbergr** [comment]
@kryzalid87 @edno 
I ended up capturing the output from Codeception with a wrapper class I was already using and checked the output for the word "password", if so I replaced the text following the comma with a message saying the password was scrubbed.  This worked because the output is generally "I <webdriver_function> <selector>, <argument>".  There are better ways to do this for sure but this was my quick and dirty way so enjoy:

`
// This function is called to run the php codeception command and intercept the output
private function exec_command($cmd, &$output) {
    if (parallel_tests > 1) {      
        // do not print by line if run in parallel
        exec($cmd, $output_array, $cmd_exit_code);      

        foreach ($output_array as $output_line) {        
            if (strpos($output_line, 'password") === false) {
                $output .= $output_line . PHP_EOL;          
            } else {          
                 // strip out passwords
                $output .= $this->scrub_password($output_line) . PHP_EOL; 
            }
        }
        echo $output;
        return $cmd_exit_code;
    }    

    // open process with pipe to STDOUT
    $process = proc_open($cmd, [1 => ['pipe", 'w"]], $pipes);
    if (!is_resource($process)) {
        throw new \Exception('Error opening process.");
    }
    // copy text from STDOUT to local var as we flush it    
    $output = '";
    while ($s = fgets($pipes[1])) {
         if (strpos($s, 'password") === false) {
            echo $s;
            flush();
            $output .= $s;
        } else {
            // strip out the actual password for security
            $scrubbed_output_line = $this->scrub_password($s);
            echo $scrubbed_output_line;
            flush();
            $output .= $scrubbed_output_line;
        }
    }
    fclose($pipes[1]);
    return proc_get_status($process)['exitcode']
}

 private function scrub_password_from_output($output_line) {
    // preg_replace catches anything after the comma, which would contain the input for fill field
    return preg_replace('/,(.*)$/", '<PASSWORD SCRUBBED>", $output_line);
}`

**Comment by kryzalid87** [comment]
@nordbergr thank you for your answer - will try this approach :)

**Comment by faust609** [comment]
@nordbergr could you please add where to use this wrapper? I've searched everywhere but I can't figure out. I'm a bit of a newbie to codeception but i've hit this problem as i have to use my password in some live tests.

**Comment by nordbergr** [comment]
@faust609 I have a php wrapper that I call which does some test set up and then calls proc_open to execute the php command to run the codeception test.  Then I use the pipes from proc_open to read the output.  Hope that is enough to get you going, if not let me know.

**Comment by unt01d** [comment]
I'm keen to see this feature too, so thought I'd have a go at [adding it](https://github.com/Codeception/Codeception/pull/4743).  Feedback appreciated!

**Comment by Naktibalda** [comment]
If you still have this issue, please look at functionality implemented by @unt01d in #4743
